docs/architecture/patterns/isolation.md

```diff
@@ -0,0 +1,153 @@
+---
+id: isolation
+sidebar_label: Isolation
+---
+
+# Isolation Pattern
+
+The Isolation pattern is a way to intercept and modify Tauri API messages sent by the frontend before they get to Tauri
+Core, all with JavaScript. The secure JavaScript code that is injected by the Isolation pattern is referred to as the
+Isolation application.
+
+## Why
+
+The Isolation pattern's purpose is to provide a mechanism for developers to help protect their application from unwanted
+or malicious frontend calls to Tauri Core. The need for the Isolation pattern rose out of threats coming from
+untrusted content running on the frontend, a common case for applications with many dependencies. See
+[Security: Threat Models](../../development/security.md#threat-models) for a list of many sources of threats that an
+application may see.
+
+The largest threat model described above that the Isolation pattern was designed in mind with was Development Threats.
+Not only do many frontend build-time tools consist of many dozen (or hundreds) of often deeply-nested dependencies, but
+a complex application may also have a large amount of (also often deeply-nested) dependencies that are bundled into the
+final output.
+
+## When
+
+The best time to use the Isolation pattern is when you have any of the `allowlist` Tauri APIs enabled. As the developer,
```

Comment:
the isolation pattern is also useful if you define custom commands right? brownfield is not as dangerous depending on the command and its inputs, but still useful